### ISSUE TITLE
[10.0] Not all product costs getting updated

### ISSUE THREAD

**Issue by matt454357** [odoo/odoo#24925]
**Impacted versions:**
10.0, 10.0e

**Steps to reproduce:**
This applies to delivery orders and manufacturing orders.  As an example:
1. Create an MO with 2 products on the BOM
1. Both products must have real time costing with FIFO
1. The first product on the bom must not be available
1. For the second product on the BOM, the standard_price must be different from the cost of the next quant to be consumed
1. Complete the MO, letting the first product result in a negative quant

**Current behavior:**
The standard_price of the second product on the BOM does not get updated.

**Expected behavior:**
The standard_price of the second product on the BOM should get updated.

**Proposed Solution:**
The problem is caused by a mistake in the function _store_average_cost_price().  It should continue when it encounters moves with negative quants.  Instead, it returns.

https://github.com/odoo/odoo/blob/3051d1dd2fee0c9d711a3f0380d967fa3f95983a/addons/stock_account/models/stock.py#L226

**Current:**
```python
if any(q.qty <= 0 for q in move.quant_ids) or move.product_qty == 0:
    # if there is a negative quant, the standard price shouldn't be updated
    return
```

**Proposed:**
```python
if any(q.qty <= 0 for q in move.quant_ids) or move.product_qty == 0:
    # if there is a negative quant, the standard price shouldn't be updated
    continue
```

**Comment by pedrobaeza** [comment]
I think you're right, @matt454357. @jco-odoo can you please check this?

**Comment by jco-odoo** [comment]
At first sight, I think so too.  I will look into it.

**Comment by pedrobaeza** [comment]
Fixed in #24944

**Comment by Yenthe666** [comment]
Good catch Pedro, done in https://github.com/odoo/odoo/commit/eabcb9ff1ae6a723fb47e340ac83a65f4f2f2400
